pkg/services/ngalert/schedule/schedule.go

```diff
@@ -476,7 +476,10 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 					isPaused := ctx.rule.IsPaused
 					// fetch latest alert rule version
 					if currentRuleVersion != newVersion {
-						if currentRuleVersion > 0 { // do not clean up state if the eval loop has just started.
+						// Do not clean up state if the eval loop has just started.
+						// We need to reset state if the loop has started and the alert is already paused. It can happen,
```

Comment:
If a brand new paused rule is file-provisioned (i.e. currentRuleVersion == 0 and isPaused = true), would there be any existing state to begin with? What gets cleaned up?

Reply:
In that particular case you mention, it won't have any state to clean up but imagine this scenario:
- Have one provisioned alert rule with, at least, one alert instance.
- We update the provisioning file to pause that alert and restart Grafana for these changes to take effect (we must have a DB that lives between restarts for this to happen)

If we don't do this change the previous-restart alert instance from the currently paused alert will linger forever for the user to see until the alert rule is updated (stale states are cleaned up in `ProcessEvalResults` which we skip when an alert is paused).

IMO, a single useless state cleanup is a better price to pay than a forever lingering useless alert instance.